pkg/api/controller_test.go

```diff
@@ -3404,45 +3404,50 @@ func TestController_ClientDisconnect(t *testing.T) {
 	clt := setupClientByEndpoint(t, server.URL, "", "")
 	_ = setupCommPrefs(t, clt)
 	cred := createDefaultAdminUser(t, clt)
-	clt = setupClientByEndpoint(t, server.URL, cred.AccessKeyID, cred.SecretAccessKey)
 
+	clt = setupClientByEndpoint(t, server.URL, cred.AccessKeyID, cred.SecretAccessKey)
 	ctx := context.Background()
-
 	_, err := deps.catalog.CreateRepository(ctx, "repo1", onBlock(deps, "repo1"), "main")
 	testutil.Must(t, err)
 
-	pr, pw := io.Pipe()
-	mpw := multipart.NewWriter(pw)
-	reqCtx, cancel := context.WithCancel(ctx)
-	go func() {
-		w, _ := mpw.CreateFormFile("content", "file.data")
-		r := rand.New(rand.NewSource(time.Now().UnixNano()))
-		_, _ = io.CopyN(w, r, 1024)
-		cancel()
-		_, _ = io.CopyN(w, r, 1024)
-		_ = mpw.Close()
-	}()
+	// prepare a client that will not wait for a response and timeout
+	dialer := net.Dialer{
```

Comment:
Didn't quite understand from the code why the client will immediately time out if dialer timeout is 30 seconds

Reply:
The dialer is used to connect as part of the transport - these are the default values.
The transport uses also the defaults, with the exception of `ResponseHeaderTimeout` it will wait for a nanoseconds for the response headers, which means it will timeout before we get the response.